server/monitor.go

```diff
@@ -3008,16 +3015,26 @@ func (s *Server) healthz(opts *HealthzOptions) *HealthStatus {
 		return health
 	}
 
-	// Check JetStream
-	js := s.getJetStream()
-	if js == nil {
+	// Check if JetStream is enabled in the config.
+	sopts := s.getOpts()
+	if !sopts.JetStream {
 		// If JetStream should be enabled then return error status.
-		if opts.JSEnabled {
+		if opts.JSEnabledOnly {
 			health.Status = "unavailable"
 			health.Error = NewJSNotEnabledError().Error()
 			return health
 		}
 		return health
+	} else if opts.JSEnabledOnly {
+		return health
+	}
+
+	// Access the Jetstream state to perform additional checks.
+	js := s.getJetStream()
+	if js == nil {
```

Comment:
With this change then `js-server-only` then would be the recommended check to use whether JetStream has been disabled/enabled during runtime

Reply:
Fixing some logic and pushing soon.. on a call with @caleblloyd :smile: